### ISSUE TITLE
Suspicious one-pixel translation in the output gradient maps

### ISSUE THREAD

**Issue by jcohenadad** [shimming-toolbox/shimming-toolbox#292]
### Checklist

- [x] I've given this issue a self-descriptive, meaningful title
- [ ] I've described my environment
- [x] I've described the bug
- [x] I've described the steps to reproduce the behaviour
- [x] I've included screenshots
- [x] I've given this issue the correct labels

### Environment details

- [ ] I followed the [installation instructions](https://shimming-toolbox.org/en/latest/2_getting_started/1_installation.html)

Project version:

Operating System and version:

Python version: 3.x.x 

Path to Python binary: /path/to/bin/python 

### Bug Description

When computing the gradient map from the fieldmap, I noticed a one-pixel translation. It looks suspicious and should be investigated further. See below:

![anim](https://user-images.githubusercontent.com/2482071/126547927-332b6dd6-c9d5-4b5b-ba6f-8f2cc65e5525.gif)

### Reproduction Steps

I ran the test procedure described here: https://github.com/shimming-toolbox/shimming-toolbox/pull/289#issue-687019550

**Comment by po09i** [comment]
I investigated both `gy` and `gz` maps. I manually calculated some pixels from the fieldmap and I get the same result as what is calculated in the phys_gradient() function. I think this is an optical illusion where it looks shifted since we are using a central difference algorithm.

Here is an example: 
Imagine `line`, a line of pixels from the image above. The second line is the gradient along that direction calculated with a central difference algorithm

```
line = [ 0,    0,    0,    1,    1,    1,    1,    0,    0,    0]
np.gradient(line)
array([ 0. ,  0. ,  0.5,  0.5,  0. ,  0. , -0.5, -0.5,  0. ,  0. ])
```

I think what is happening is since the 0.5 gradient is negative and the other positive, one becomes darker and the other whiter which tricks us into seeing a translation. This is what I think is happening with all the "noise" pixels ([0,1,0] --> [1, 0,-1]) and the white spot at the bottom of the fieldmap. Let me know what you think.
